src/Microsoft.Data/BaseDataFrameColumn.cs

```diff
@@ -0,0 +1,25 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Microsoft.Data
+{
+    public abstract class BaseDataFrameColumn
+    {
+        public BaseDataFrameColumn(string name, long length = 0)
+        {
+            Length = length;
+            Name = name;
+        }
+        public long Length { get; protected set; }
+        public long NullCount { get; protected set; }
+        public string Name;
+
+        public virtual object this[long rowIndex] { get { throw new NotImplementedException(); } set { throw new NotImplementedException(); } }
```

Comment:
This is temporary code, but long term it should be on several lines (using => syntax if it's oneliners)